### ISSUE TITLE
🤖 [AUTO] Implement Strategy: spx_components_normalized_returns_regression_tree

### ISSUE THREAD

**Issue by kpakpooluwaseyi** [kpakpooluwaseyi/Agentic-Algo-Strategies#320]
## 🤖 Auto-Generated Strategy Request

**Source:** Artificial Intelligence Techniques 2017.pdf (strategy 9)
**Extracted:** 2025-12-16T14:18:19.776616

---

1.  **Strategy Name**: spx_components_normalized_returns_regression_tree
2.  **Strategy Type**: Cross-sectional / Quantitative/Machine Learning
3.  **Timeframe**: 1D (daily close to daily close)
4.  **Instruments**: SPX Component Stocks (survivorship-bias-free, using midprice at market close)

5.  **Entry Rules**:
    *   **Long**: Predicted next-day normalized return for a specific SPX component stock is positive.
    *   **Short**: Predicted next-day normalized return for a specific SPX component stock is negative.

6.  **Exit Rules**:
    *   **Time-based**: Hold position for one day.
    *   **Take Profit/Stop Loss**: Not explicitly mentioned, implied by one-day hold.

7.  **Indicators Required**:
    *   Normalized Past 1-day return (`ret1N = ret1 ./ vol1`)
    *   Normalized Past 2-day return (`ret2N = ret2 ./ vol1`)
    *   Normalized Past 5-day return (`ret5N = ret5 ./ vol1`)
    *   Normalized Past 20-day return (`ret20N = ret20 ./ vol1`)
    *   `vol1`: Past daily return volatility (used for normalization of both predictors and response).
    *   Cross-validated Regression Tree model (e.g., MATLAB's `fitrtree` with `KFold=5`, `MinLeafSize` 100) trained on aggregated and normalized data from all SPX component stocks.

8.  **Risk Management**:
    *   **Position Sizing**: Not explicitly detailed, but implied to involve buying/shorting based on signals. The normalization of returns helps manage risk across different stocks.
    *   **Maximum Risk per Trade**: Not explicitly defined.
    *   **Risk-Reward Ratio**: Not explicitly defined.

9.  **Additional Notes**:
    *   Training data involves aggregating historical data for hundreds of SPX component stocks into a single dataset for the machine learning algorithm.
    *   Data range: January 3, 2007, to December 31, 2013, for the training set.
    *   Out-of-sample performance: CAGR 2.3%, Sharpe ratio 0.9.
    *   Normalization of returns by their past volatility is critical for aggregating cross-sectional data, as unnormalized data led to significantly worse performance (CAGR -0.7%, Sharpe -0.4).

---

## Implementation Instructions

Please implement this strategy using the `backtesting.py` framework with the following requirements:

1. **File Location:** `strategies/spx_components_normalized_returns_regression_tree.py`
2. **Data Path:** Use `data/BTC-USD-15m.csv` for backtesting
3. **Indicators:** Use `talib` or `pandas_ta` (NOT backtesting.py's built-in indicators)
4. **Template:** Follow existing strategies in the `strategies/` folder

### Required Components:
- Strategy class inheriting from `Strategy`
- Proper indicator initialization using `self.I()`
- Entry/exit logic in `next()` method
- Default parameters as class attributes
- Risk management (stop loss, take profit)

### Data Format:
```
datetime, open, high, low, close, volume
2023-01-01 00:00:00, 16531.83, 16532.69, 16509.11, 16510.82, 231.05
```

### Output:
- Run backtest with default parameters
- Print stats and create HTML plot
- Save results to `results/temp_result.json`
